src/codewhisperer/service/recommendationHandler.ts

```diff
@@ -148,27 +142,25 @@ export class RecommendationHandler {
         autoTriggerType?: CodewhispererAutomatedTriggerType,
         pagination: boolean = true,
         page: number = 0
-    ) {
+    ): Promise<GetRecommendationsResponse> {
+        let invocationResult: 'Succeeded' | 'Failed' = 'Failed'
+        let errorMessage: string | undefined = undefined
+
         if (!editor) {
-            return
+            return Promise.resolve<GetRecommendationsResponse>({
+                result: invocationResult,
+                errorMessage: errorMessage,
```

Comment:
nit: errorMessage here is `undefined`, make it `no active editor found`?

Reply:
Making it a non undefined will make it show a notification for this error, which changes the current behavior. When the user closes the current editor, I think we can just sliently exit without popuping a message.